src/decoder/zlib.rs

```diff
@@ -101,10 +111,9 @@ impl ZlibStream {
             return Ok(());
         }
 
-        loop {
+        while !self.state.is_done() {
```

Comment:
I've tweaked the loop to get the `debug_assert!(!self.state.is_done()` inside `prepare_vec_for_appending`.